apps/backend/src/shared/infra/http/middlewares/check-parent-relation.ts

```diff
@@ -0,0 +1,85 @@
+import { NextFunction } from 'express'
+
+import { AuthenticatedMedusaRequest, MedusaResponse } from '@medusajs/framework'
+import {
+  ContainerRegistrationKeys,
+  MedusaError
+} from '@medusajs/framework/utils'
+
+type CheckChildParentRelation<Body> = {
+  parentResource: string
+  childField: string
+  childId: (req: AuthenticatedMedusaRequest<Body>) => string
+  parentId?: (req: AuthenticatedMedusaRequest<Body>) => string
+}
+
+/**
+ * Middleware that verifies if the specified entities are in database relationship.
+ *
+ * @param options - Configuration options for the check
+ * @param options.parentResource - The parent entity type
+ * @param options.childField - Field to select that indicates the relation
+ * @param options.childId - Request parameter containing the child resource ID
+ * @param options.parentId - Request parameter containing the parent resource ID (defaults to 'id')
+ *
+ * @throws {MedusaError} If the child does not belong to the parent resource
+ *
+ * @example
+ * // Basic usage - check if variant belongs to product
+ * app.use(checkChildParentRelation({
+ *   parentResource: 'product',
+ *   childField: 'variants.id',
+ *   childId: (req) => req.params.variantId
+ * }))
+ *
+ * @example
+ * // Custom field usage - check if inventory item belongs to variant
+ * app.use(checkChildParentRelation({
+ *   parentResource: 'variant',
+ *   childField: 'inventory_items.inventory_item_id',
+ *   childId: (req) => req.params.inventoryItemId,
+ *   parentId: (req) => req.params.variantId
+ * }))
+ */
+export const checkChildParentRelation = <Body>({
```

Comment:
remove this you can just check product variants based on product id and passing product id to the selector

```ts
// check whether product_id is related to logged in seller
const selector = { id: variant_id, product_id }
updateProductVariant(selector, req.body)
```